polaris-core/src/main/java/org/apache/polaris/core/persistence/resolver/PolarisResolutionManifest.java

```diff
@@ -277,6 +285,7 @@ public PolarisResolvedPathWrapper getResolvedRootContainerEntityAsPath() {
 
   public PolarisResolvedPathWrapper getResolvedReferenceCatalogEntity(
       boolean prependRootContainer) {
+    requirePrimaryResolverStatus();
```

Comment:
[doubt] is just checking status not null sufficient ? or should we bake the assertion that that status != null and status = SUCCESS ?